Setup (more) generic form helpers

annavik opened the pull request:
We already have some form helpers from before (see `FormController` and `FormField`), but in this PR we will also setup some help components to handle form layout. Also we setup a hook `useFormError` to streamline error handling further.

As part of this PR we are also updating current forms to use these new helpers. In the next one, we will create som new forms for projects. Curious to see if we can do that without adding a lot of code, thanks to this.

mihow reviewed (changes requested):
Awesome! It looks seamless, except for I am unable to open the dialog for "register new deployment"

`Uncaught TypeError: Cannot read properties of undefined (reading 'defaultPrevented')`

<img width="620" alt="image" src="https://github.com/RolnickLab/ami-platform/assets/158175/acff44a2-ad14-42d9-a774-e7fab8358188">

annavik commented:
> Awesome! It looks seamless, except for I am unable to open the dialog for "register new deployment"
> 
> `Uncaught TypeError: Cannot read properties of undefined (reading 'defaultPrevented')`
> 
> <img alt="image" width="620" src="https://user-images.githubusercontent.com/158175/272783757-acff44a2-ad14-42d9-a774-e7fab8358188.png">

Oops! I had forgot to sync with main where I fixed this

Also added some new form stuff for delete, in that case we probably want to have a complete form to reuse
